numpy/core/src/multiarray/stringlib/add.h

```diff
@@ -0,0 +1,40 @@
+#ifndef STRINGLIB_LENGTH_H
+#error must include "stringlib/length.h" before including this module
+#endif
+
+static inline int
+STRINGLIB(add)(PyArrayIterObject **in_iters, PyArrayIterObject *out_iter)
+{
+    STRINGLIB_CHAR *left, *right;
+    Py_ssize_t left_len, right_len, left_len_in_bytes, right_len_in_bytes;
+
+    left = (STRINGLIB_CHAR *) in_iters[0]->dataptr;
+    right = (STRINGLIB_CHAR *) in_iters[1]->dataptr;
+
+    left_len = STRINGLIB(get_length)(in_iters[0]);
+    left_len_in_bytes = left_len * sizeof(STRINGLIB_CHAR);
+
+    right_len = STRINGLIB(get_length)(in_iters[1]);
+    right_len_in_bytes = right_len * sizeof(STRINGLIB_CHAR);
+
+    if (left_len + right_len > PY_SSIZE_T_MAX) {
```

Comment:
If this overflows, its negative.  So this doesn't guard against anything, I think?!
You could use `size_t`, I suppose.  (In which case, overflows are not even possible for `add`, since even if we bump limit, they would be bumped to `ssize_t` and `size_t` is at least double that range)

Reply:
Oh yeah, that's correct. Fixed in https://github.com/numpy/numpy/pull/24756/commits/cea8f067360993ed9070db130246cac71ae0545e by not doing the addition at all.